src/main/java/appeng/block/AEBaseBlock.java

```diff
@@ -73,41 +59,32 @@
 import appeng.core.features.FeatureNameExtractor;
 import appeng.core.features.IAEFeature;
 import appeng.core.features.IFeatureHandler;
-import appeng.core.features.ItemStackSrc;
 import appeng.helpers.AEGlassMaterial;
 import appeng.helpers.ICustomCollision;
-import appeng.tile.AEBaseTile;
-import appeng.tile.networking.TileCableBus;
-import appeng.tile.storage.TileSkyChest;
 import appeng.util.LookDirection;
 import appeng.util.Platform;
-import appeng.util.SettingsFrom;
 
 
-public class AEBaseBlock extends BlockContainer implements IAEFeature
+public abstract class AEBaseBlock extends Block implements IAEFeature
 {
 	private final String featureFullName;
-	private final Optional<String> featureSubName;
+	protected final Optional<String> featureSubName;
 	@SideOnly( Side.CLIENT )
 	public IIcon renderIcon;
 	protected boolean isOpaque = true;
 	protected boolean isFullSize = true;
 	protected boolean hasSubtypes = false;
 	protected boolean isInventory = false;
+	protected IFeatureHandler handler;
 	@SideOnly( Side.CLIENT )
 	BlockRenderInfo renderInfo;
```

Comment:
rather loosening the fields contract, better use `private` on all and use `final` getter and setter where required.